src/offset_of.rs

```diff
@@ -145,4 +173,25 @@ mod tests {
 
         assert_eq!(offset_of!(sub::Foo, x), 0);
     }
+
+    #[test]
+    fn generic_offset() {
+        #[repr(C, packed)]
+        struct Foo<T, U>(T, U);
+        let o = offset_of!(Foo<u8, u8>, 1);
```

Comment:
This is not at all the testcase I suggest! In my testcase, `offset_of` was used on a type that uses generics from the environments. `Foo<u8, u8>` does not do that, it is a monomorphized instance of a generic type.

Reply:
Oh, I see, that wasn't clear

Reply:
Is that something we're okay with? Is that a temporary restriction?

Reply:
So you verified that it indeed not working?

Reply:
Yes, it indeed does not work